pkg/client/cloudevents/source_client.go

```diff
@@ -43,6 +44,12 @@ func NewSourceClient(sourceOptions *ceoptions.CloudEventsSourceOptions, resource
 			logger.Infof("received action %s for resource %s", action, resource.ID)
 			switch action {
 			case cetypes.StatusModified:
+				// if the consumer is not owned by this instance, ignore it
+				if !dispatcher.Dispatch(resource.ConsumerID) {
```

Comment:
note to myself, the Dispatch should be threadsafe.

Reply:
The consumerSet is designed as a threadsafe set. During the Dispatch operation, the contains method is invoked on the set, incorporating a read lock for secure access. see: https://github.com/deckarep/golang-set/blob/main/set.go#L201-L207 and https://github.com/deckarep/golang-set/blob/main/threadsafe.go#L61C1-L67C2